apps/src/lib/script-editor/ScriptEditor.jsx

```diff
@@ -454,16 +455,23 @@ export default class ScriptEditor extends React.Component {
           />
         </div>
         <h2>Stages and Levels</h2>
-        <div>
-          <textarea
-            id="script_text"
-            name="script_text"
-            rows={textAreaRows}
-            style={styles.input}
-            defaultValue={this.props.stageLevelData || "stage 'new stage'\n"}
-            ref={textArea => (this.scriptTextArea = textArea)}
-          />
-        </div>
+        {this.props.beta ? (
+          <FlexGroup />
+        ) : (
+          <div>
+            <a href="?beta=true">
+              Try the beta Script Editor (will reload the page without saving)
```

Comment:
Is there a way to switch back and forth without a full page reload? I think this warning was one of the barriers to adoption.

Reply:
react could handle switching the UI nicely, but the hard part seems like it would be switching the "source of truth" back and forth between raw text (in the old editor) and redux state representing the actual stage hierarchy (in the new editor). you could make changes in the new ui and use that to update the old ui via serializing, but it would be a lot of new, throwaway code to get the old editor ui to be able to update the new redux state.

the way I'm imagining rolling this out is once we have feature parity, put it up on an adhoc and get curriculum folks to bash on it for a bit. if it looks good, make a PR to have the UI replace the text interface. so, no real period of keeping both editing experiences around and active.

Reply:
Got it, makes sense!